README.md: add guidance on expressions of interest

odscjen opened the pull request:
closes https://github.com/open-contracting/ocds-extensions/issues/41

odscjen committed:
README.md: add guidance on using extension for pre-qual responses.

odscjen committed:
README.md: update line 13

odscjen committed:
Merge branch 'master' into 41_pre-qual_guidance

odscjen committed:
README.md: change request to participate to expression of interest as per #1649

jpmckinney committed:
Merge branch 'master' into 41_pre-qual_guidance

jpmckinney committed:
readme: Copy-edit guidance on expressions of interest

jpmckinney committed:
Merge branch 'master' into 41_pre-qual_guidance

jpmckinney committed:
Merge branch 'master' into 41_pre-qual_guidance

jpmckinney committed:
metadata: Update extension name and description

jpmckinney committed:
Merge branch 'master' into 41_pre-qual_guidance

duncandewhurst commented on README.md:
````diff
@@ -215,6 +221,58 @@ A potential supplier submits a bid for two lots as a single document:
+        "description": "Expression of interest submission detailing how the bidder meets the eligibility criteria laid out in the pre-qualification phase notice.",
+        "tenderers": [
+          {
+            "id": "MEGA",
+            "name": "Mega Consortium"
+          }
+        ],
+        "countriesOfOrigin": [
+          "MX"
+        ],
+        "documents": [
+          {
+            "id": "doc-1",
+            "documentType": "eligibilityCriteria",
+            "title": "Expression of interest - evidence of eligibility."
+          }
+        ]
+      },
+      {
+        "id": "2",
+        "date": "2016-12-10T01:00:00+01:00",
+        "status": "pending",
+        "tenderers": [
+          {
+            "id": "BETA",
+            "name": "Beta Consortium"
+          }
+        ],
+        "documents": [
+          {
+            "id": "doc-2",
+            "documentType": "eligibilityCriteria",
+            "title": "Expression of interest - evidence of eligibility."
+          }
+        ]
+      }
+    ]
+  }
+}
+```
````
Since there's presently no difference in modelling bids and EOI responses, I think this example can be removed.

If the discriminator field proposed in https://github.com/open-contracting/ocds-extensions/issues/41#issuecomment-1916477736 is added, we can add a minimal example to demonstrate the usage of that field.

```suggestion
```

odscjen replied:
agree that a 3 example is unnecessary, I've removed this and added the new discriminator field to the other examples, turning the one that's just about submissions into an EOI submission.

duncandewhurst reviewed (commented):
I think that we should pause this PR until the points discussed in https://github.com/open-contracting/ocds-extensions/issues/41 are settled (whether to add a discriminator field, whether to rename `Bid` to `Response` etc.) as that will likely impact the content of the documentation and examples.

jpmckinney commented:
> I think that we should pause this PR until the points discussed in https://github.com/open-contracting/ocds-extensions/issues/41 are settled

I'll follow up - thanks for the reminder.

jpmckinney commented:
> whether to add a discriminator field, whether to rename Bid to Response etc.

Let's add a discriminator field and do the renaming, but we can discuss details in https://github.com/open-contracting/ocds-extensions/issues/41

odscjen committed:
add submissionType and codelist and update examples

odscjen committed:
Merge branch 'master' into 41_pre-qual_guidance

jpmckinney commented on README.md:
```diff
@@ -44,6 +46,10 @@ If a potential supplier submits a bid for multiple lots as a single document, fo
 
 If the bid cannot be divided (for example, the data source describes only the total value of the bid, and not the individual value for each lot within the bid), create one `Bid` object, and add all lots' identifiers to the object's `relatedLots`.
 
+### Publish expressions of interest
```
Verbs are generally good, but "publish" would be too frequent, so we omit it.

```suggestion
### Expressions of interest
```

jpmckinney commented on README.md:
```diff
@@ -44,6 +46,10 @@ If a potential supplier submits a bid for multiple lots as a single document, fo
 
+### Publish expressions of interest
+
+Buyers and procuring entities can publish expressions of interest, which are received during pre-qualification or pre-selection, in the `bids.details` array. Use the `bids.details.submissionType` field to indicate if the published data refer to a bid or an expression of interest.
```
Buyers do not do the publishing. If we want to describe expressions of interest, that should be in the intro or in the schema sections of the readme.

```suggestion
Expressions of interest are also disclosed in the `bids.details` array. Use the `bids.details.submissionType` field to indicate whether a submission is a bid or an expression of interest.
```

jpmckinney commented on README.md:
````diff
@@ -158,7 +166,7 @@ Aggregate post-award statistics and individual bid submissions:
 ```
 
-A potential supplier submits a bid for two lots as a single document:
+A potential supplier submits an expression of interest for two lots as a single document:
````
I think it's unusual for an expression of interest to only describe the price, so let's keep it as a bid example.

odscjen replied:
I think this example is useful for the case of bids on multiple lots, but if it's to stay a bid (which makes a lot of sense) should we add a third example for an EOI?

jpmckinney replied:
I think the submissionType is simple enough to not need an example

odscjen replied:
I agree, great I'll leave it as it then

jpmckinney committed:
readme: Fix changelog order (bad merge)

jpmckinney reviewed (commented):
Need to also modify the title and description of the Bid definition.

It would be difficult to modify its machine-name to Submission, as other extensions modify the Bid definition (requirements, subcontracting).

Edit:

### Schema

Need to add "expression of interest" wording to `bids` (reword as "Information about ..." and copy the wording of the extension metadata), `Bids` (and fields), `Bid` fields, etc. Can maybe consider "submission" to reduce wordiness. Please address style guide issues like "An array of" while making edits.

### bidStatus

Need to probably also modify bidStatus to use "bid or expression of interest". Noting that "evaluated" can be used both in the context of qualification criteria and award criteria, e.g. from [UNCITRAL](https://uncitral.un.org/sites/uncitral.un.org/files/media-documents/uncitral/en/2011-model-law-on-public-procurement-e.pdf):

> The procuring entity shall evaluate the qualifications of suppliers or con- tractors in accordance with the qualification criteria and procedures set out in the pre-qualification or pre-selection documents

That said, we probably need to remove "in order to be evaluated" as that's being used to mean "against award criteria" which isn't the case for EOIs.

We can also update "requirements" to "criteria".

### statistic

For consistency, in `statistic.csv`, we should change requests to participate to expressions of interest. We can leave the `requests` code as-is as it has been present since the first version.

Change the Category for `requests` to `expressionsOfInterest`

jpmckinney committed:
readme: Add more references to expressions of interest. Copy-edit.

- bids array is not only for received submissions (e.g. can be invited)
- Merge related paragraphs

odscjen committed:
Apply suggestions from code review

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

odscjen commented:
@jpmckinney Re. `statistic` should we also add new codes to disclose the results of the evaluation of the EOIs? So:
* 'validExpressionsOfInterest'
* 'disqualifiedExpressionsOfInterest'

odscjen committed:
review suggests re style guide field descriptions

jpmckinney commented:
> disclose the results of the evaluation of the EOIs

If we didn’t need such codes for eForms, then I think no jurisdiction tracks such statistics.

odscjen committed:
Merge branch 'master' into 41_pre-qual_guidance

jpmckinney committed:
readme: Move changelog entry, and unlist changes to newly added codes

jpmckinney committed:
bidStatus: Remove "listed" to be consistent with withdrawn code

jpmckinney committed:
release-schema: Parenthesize "(or expressions of interest)" as awards and contracts should relate to bids, and only to EOIs if bids are not published

jpmckinney committed:
release-schema: Eliminate unnecessary definition

jpmckinney committed:
release-schema: Remove wording that asks publisher to be consistent (would be too repetitive to do this consistently)

jpmckinney committed:
release-schema: Use "summary statistics" instead of "aggregate statistics", consistently

jpmckinney committed:
release-schema(Statistic): Omit sentence relating to a specific field in the definition's description.

jpmckinney committed:
release-schema(bids/details): Avoid listing fields. Omit sentence that is true for any field.

jpmckinney committed:
release-schema: Use "submission" in titles

jpmckinney committed:
release-schema: Remove use of "qualification stage" which does not exist in OCDS.

jpmckinney committed:
release-schema: Add missing "expression of interest" in `identifiers`. Parenthesize it in `value`, as EOIs mostly don't have values.

jpmckinney committed:
release-schema(bids/statistics): Match readme wording. readme: Fix changelog entry to match heading.

jpmckinney reviewed (approved):
I made some other changes. Ready to merge.

jpmckinney committed:
readme: Fix references to bids.details and .statistics
